Add pages for HTML*Element.willValidate (#35665)
mdn/content: files/en-us/web/api/htmlbuttonelement/willvalidate/index.md

---
title: "HTMLButtonElement: willValidate property"
short-title: willValidate
slug: Web/API/HTMLButtonElement/willValidate
page-type: web-api-instance-property
browser-compat: api.HTMLButtonElement.willValidate
---

{{APIRef("HTML DOM")}}

> The **`willValidate`** read-only property of the {{domxref("HTMLButtonElement")}} interface indicates whether the button is a candidate for [constraint validation](/en-US/docs/Web/HTML/Constraint_validation). It is `false` if any conditions bar it from constraint validation, including: its {{domxref("HTMLButtonElement.type", "type")}} is `reset` or `button`; it has a {{HTMLElement("datalist")}} ancestor; or the {{domxref("HTMLButtonElement.disabled", "disabled")}} property is set to `true`.

Reviewer: ```suggestion
The **`willValidate`** read-only property of the {{domxref("HTMLButtonElement")}} interface indicates whether the {{htmlelement("button")}} is a candidate for [constraint validation](/en-US/docs/Web/HTML/Constraint_validation). It is `false` if any conditions bar it from constraint validation, including being of {{domxref("HTMLButtonElement.type", "type")}} `reset` or `button`, having a {{HTMLElement("datalist")}} ancestor, or its {{domxref("HTMLButtonElement.disabled", "disabled")}} property is `true`.
```

* removed "set to" to make it clearer that the `disabled` attribute also mean no constraint validation.
* the semi-colons were accurate grammar-wise, but suggest going with commas (my suggested text) or an unordered list starting with "including:" ....
